dependencies/sqlite/meson.build

```diff
@@ -0,0 +1,16 @@
+# SQLite Dependency File
+#============================================================================
+
+
+# Look for sqlite3, a required dependency.
+#--------------------------------------------------
+sqlite3_dep = declare_dependency(
+    dependencies: cpp.find_library('sqlite3'),
```

Comment:
sqlite3 has a pkg-config, why not use `dependency()` here?

Reply:
iirc it was an issue on BSD

Reply:
Can you confirm that? We have several in-meson wrappers for hanlding cases of "*BSD or MacOS have this as a part of their base os without a pkg-config file", zlib being an example. We could do the same for sqlite if that's the case.

Though, generally the preferred way to do this is:
```meson
sqlite3_dep = dependency('sqlite3', required : false)
if not sqlite3_dep.found()
  sqlite3_dep = cpp.find_library('sqlite3')
endif
```

Reply:
i'm not 100%, i just remember it was causing a massive pain on freeBSD, so i switched to a library decl

Reply:
Maybe some bug to be solved by upstream Meson?

Reply:
earlier i went back and compiled nix on bsd again. it seems to be an issue with meson not correctly searching `/usr/local`, as when you manually modify the sqlite install to move it to `/usr`, it compiles just fine.